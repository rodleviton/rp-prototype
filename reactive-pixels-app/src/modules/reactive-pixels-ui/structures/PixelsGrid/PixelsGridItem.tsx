import classnames from "classnames";
import { css } from "emotion";
import * as React from "react";
import { Box } from "../../components/Box";
import { IBaseTheme, withStyles } from "../../theme";

interface IClasses {
  root: string;
}

interface IProps {
  classes: IClasses;
  className?: string;
}

const styles = (theme: IBaseTheme): IClasses => {
  return {
    root: css({
      margin: 20
    })
  };
};

/**
 * PixelsGridItem
 *
 * @param {string} [className=""] - Extend css classes to override default styling.
 * @example
 * <PixelsGridItem />
 */
export class PixelsGridItem extends React.PureComponent<IProps> {
  public render() {
    const { classes, className, children, ...otherProps } = this.props;

    return (
      <Box className={classnames(classes.root, className)} {...otherProps}>
        {children}
      </Box>
    );
  }
}

export default withStyles(styles)(PixelsGridItem);
