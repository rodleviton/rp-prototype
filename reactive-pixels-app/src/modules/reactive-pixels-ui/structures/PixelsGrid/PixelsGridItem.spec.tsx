import * as React from "react";
import { render } from "react-testing-library";
import WithRouter from "../../utils/test/WithRouter";
import WithTheme from "../../utils/test/WithTheme";
import PixelsGridItem from "./PixelsGridItem";

describe("<PixelsGridItem />", () => {
  it("should render <PixelsGridItem /> component", () => {
    const { container } = render(
      <WithRouter>
        <WithTheme>
          <PixelsGridItem />
        </WithTheme>
      </WithRouter>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
