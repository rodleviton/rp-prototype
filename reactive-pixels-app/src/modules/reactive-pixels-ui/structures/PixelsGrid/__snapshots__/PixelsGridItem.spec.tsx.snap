// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PixelsGridItem /> should render <PixelsGridItem /> component 1`] = `
<div
  class="css-0 css-zwiamn sc-bdVaJa bDWFJH"
/>
`;
